### PR TITLE
Integrate self-efficacy questionnaire

### LINKED ISSUE #14: Add self-efficacy questionnaire
I need to integrate the forgotten self-efficacy questionnaire.
The good thing is that I can automatically redirect users to it at the beginning and at the end of the experiment :)

### PR DESCRIPTION
**Pull request by Ziip-dev**
This PR integrates the questionnaire app into the main website

### CODE REVIEW

**Comment by Ziip-dev** [comment]
Let's create the questionnaire app first

**Comment by Ziip-dev** [comment]
What is required:
- [ ] questionnaire model
- [ ] questionnaire template
- [ ] questionnaire view

File: dashboard/views.py [hunk 1]
```diff
@@ -28,14 +29,23 @@ def home(request):
     :return: The home page template or the login template.
     """
     if request.user.is_authenticated and is_integrated(request.user):
-        #  If not already present, store the experiment start date after the first login
+        #  Store the experiment start date after the first login and redirect to questionnaire
         if not StartDate.objects.filter(user=request.user).exists():
             StartDate(user=request.user).save()
+            # redirect to self-efficacy questionnaire
```
Comment (Ziip-dev) [inline dashboard/views.py:35]: This is the first entry point of the questionnaire.
We must hit it at the beginning of the experiment.

File: dashboard/views.py [hunk 2]
```diff
@@ -28,14 +29,23 @@ def home(request):
...
+        # Retrieve user start date as datetime object and compute end_date
         start_date = StartDate.objects.get(user=request.user).start_date
+        end_date = start_date + timedelta(days=15)
+        print("""
+              start_date = {start_date}
+              today      = {date.today()}
+              end_date   = {end_date}
+              """)
+        # if date.today() == end_date:
+        # redirect to self-efficacy questionnaire
```
Comment (Ziip-dev) [inline dashboard/views.py:46]: This is the second entry point of the questionnaire.
We must hit it after two weeks of experiment.

File: dashboard/views.py [hunk 3]
```diff
@@ -28,14 +27,25 @@ def home(request):
...
         start_date = StartDate.objects.get(user=request.user).start_date
+        end_date = start_date + timedelta(days=15)
+        print(
+            """
+              start_date = {start_date}
+              today      = {date.today()}
+              end_date   = {end_date}
+              """
+        )
+        # if date.today() == end_date:
```
Comment (Sundava) [inline dashboard/views.py:45]: Dead code as comments is ugly

File: questionnaire/fixtures/questions.json [hunk 4]
```diff
@@ -0,0 +1,86 @@
+[
```
Comment (Sundava) [inline questionnaire/fixtures/questions.json:1]: Guessing this won't be up to date with the changes to the `Question` model (namely, the presence of the `questionnaire` Foreign Key

File: questionnaire/views.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#     success_url = reverse_lazy("questionnaire")
+#     # fields = ["question", "answer"]
+#     # context_object_name = "answers"
+
+#     def get_success_url(self, **kwargs):
+#         pass
+
+
+# ANTOINE
+class QuestionnaireView(CreateView):
```
Comment (Sundava) [inline questionnaire/views.py:50]: What I wrote was heavy pseudocode, you'll probably at least need to do something with 
`def get_success_url`, and ensure the returned url points to the next question in the questionnaire
```python
 return reverse_lazy("questionnaire". args={"question" : int(kwargs['question'])+1})
```

File: dashboard/views.py (hunk 3 above)
Comment (Ziip-dev) [reply dashboard/views.py:45]: This will be uncommented at the end of the PR

File: questionnaire/fixtures/questions.json (hunk 4 above)
Comment (Ziip-dev) [reply questionnaire/fixtures/questions.json:1]: Done

File: run-webserver.sh [hunk 6]
```diff
@@ -8,6 +8,7 @@ source /home/ziip/pysetup/.venv/bin/activate
 # Update webserver - migrations and staticfiles
```
[the remark this answers is not in the document]
Comment (Ziip-dev) [reply run-webserver.sh:8]: This is the entrypoint for the [Dockerfile](https://github.com/Ziip-dev/SQS_HBS/blob/d605ccbb2b1aed967fd0897438940aa86d8c4471/webserver.Dockerfile#L95), so the path refers to the user's home directory created within the image

File: questionnaire/views.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    model = Answer
+    form_class = QuestionnaireForm
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        questionnaire = Questionnaire.get(pk=kwargs["questionaire_pk"])
```
[the remark this answers is not in the document]
Comment (Ziip-dev) [reply questionnaire/views.py:56]: After testing they are :) Retrieving them was through
```
self.kwargs["questionnaire_pk"]
```

File: questionnaire/views.py (hunk 5 above)
Comment (Ziip-dev) [reply questionnaire/views.py:50]: I sorted that out like this:
```python
    def get_success_url(self):
        question_id = self.kwargs["question"] + 1
        questionnaire_pk = self.kwargs["questionnaire_pk"]
        max_question = Question.objects.filter(
            questionnaire_id=questionnaire_pk
        ).count()

        # Return to questionnaire with the next question
        if question_id < max_question:
            return reverse_lazy(
                "questionnaire",
                kwargs={"questionnaire_pk": questionnaire_pk, "question": question_id},
            )

        # Return to home when all the questions have been answered
        elif question_id == max_question:
            return reverse_lazy("home")
```

Not sure this is neither the cleanest nor best practices way to do it but that works.
Advice is welcome if you have a better approach in mind :)

**Pull request merged**